library/core/src/num/mod.rs

```diff
@@ -791,7 +793,7 @@ impl u8 {
     #[rustc_const_stable(feature = "const_ascii_ctype_on_intrinsics", since = "1.47.0")]
     #[inline]
     pub const fn is_ascii_alphanumeric(&self) -> bool {
-        matches!(*self, b'0'..=b'9') | matches!(*self, b'A'..=b'Z') | matches!(*self, b'a'..=b'z')
+        matches!(*self, b'0'..=b'9' | b'A'..=b'Z' | b'a'..=b'z')
```

Comment:
Why this unrelated change? this introduces branch https://rust.godbolt.org/z/8onesWshb